cirq/experiments/fidelity_estimation.py

```diff
@@ -642,3 +646,195 @@ def per_cycle_depth(df):
         return pd.Series({'fidelity': fid_lsq})
 
     return df.reset_index().groupby('cycle_depth').apply(per_cycle_depth).reset_index()
+
+
+# mypy issue: https://github.com/python/mypy/issues/5374
+@dataclass(frozen=True)  # type: ignore
+class XEBPhasedFSimCalibrationOptions:
+    """Options for calibrating a PhasedFSim-like gate using XEB.
+
+    You may want to use more specific subclasses like `SqrtISwapXEBOptions`
+    which have sensible defaults.
```

Comment:
Instead of creating subclasses for defaults, I would crate a creation method that creates instance with correct values. I would also assume that all the parameters must be provided explicitly. The defaults here are meaningless and impractical, right?

Reply:
So I'm going through and re-writing this without subclassing but I'm worried I'm just inventing the notion of subclasses. The rough sketch is 

 - move `should_parameterize` to a callable class attribute
 -  write a factory function that fills in the correct values

The factory function--however--should still accept all the characterize_xxx bools *and also* most of the `xxx_default` floats and override the defaults inherent in the factory function. The reason why we still need to override is because we still want people to choose whether or not they're doing certain angles and we may have additional information about a good starting point for optimizing the angles (for example: theory that phi is a small non-zero number or results from a previous characterization)

At this point my factory function is encoding a lot of the behaviors of subclassing; and passing a callable around is similar to overriding an abstract method.

Reply:
I missed the existence of `should_parametrize`, this is in fact a strong pointer for subclassing.

As for defaults, I'm still not really convinced but I'm not pushing for this as well. If there are defaults which you think are reasonable than it would be good to have a create method with name that describes what those defaults are. For example, the current defaults would be something like `create_identity_gate_with_all_angles_options`. Again, this is something which will become more clear later I guess

Reply:
The defaults are used to seed the classical optimizer. *In theory* you should be able to find the true angles starting from any initial guess, it might just take longer. In practice it could send you into a different local optimum. The defaults correspond to what gate you're replacing/parameterizing; which is why setting the default-defaults currently goes in the subclass (with the `should_parameterize` override)